dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report-sql.repo.ts

```diff
@@ -1,11 +1,31 @@
 // TODO FN-1853 - rename this to `utilisation-report.repo.ts` when all repo
 //  methods have been migrated from MongoDB to SQL
 import { SqlDbDataSource } from '@ukef/dtfs2-common/sql-db-connection';
-import { UtilisationReportEntity } from '@ukef/dtfs2-common';
-import { ReportPeriod } from '../../types/utilisation-reports';
+import { UtilisationReportEntity, ReportPeriod, UtilisationReportReconciliationStatus } from '@ukef/dtfs2-common';
 
 export const UtilisationReportRepo = SqlDbDataSource.getRepository(UtilisationReportEntity).extend({
   async findOneByBankIdAndReportPeriod(bankId: string, reportPeriod: ReportPeriod): Promise<UtilisationReportEntity | null> {
     return await this.findOneBy({ bankId, reportPeriod });
   },
+  async findOpenReportsBeforeReportPeriodStartForBankId(bankId: string, reportPeriodStart: ReportPeriod['start']): Promise<UtilisationReportEntity[] | null> {
+    const statusToExclude: UtilisationReportReconciliationStatus = 'RECONCILIATION_COMPLETED';
+    return await this.createQueryBuilder('utilisationReport')
+      .where('utilisationReport.bankId = :bankId', { bankId })
+      .andWhere('utilisationReport.status != :statusToExclude', { statusToExclude })
+      .andWhere(
+        this.createQueryBuilder('utilisationReport')
+          .where('utilisationReport.reportPeriod.start.year < :reportPeriodStart.year', { reportPeriodStart })
+          .orWhere(
+            'utilisationReport.reportPeriod.start.year = :reportPeriod.start.year AND utilisationReport.reportPeriod.start.month < :reportPeriodStart.month',
+            { reportPeriodStart },
+          ),
+      )
+      .getMany();
+    // TODO FN-1862 should we install the typeorm dependency to use that instead of query builder?
+    // return await this.find({
+    //   where: {
+    //     bankId: Equal
+    //   }
+    // })
```

Comment:
There are things exported by `typeorm` which can do this sort of logic. However, we don't have `typeorm` as a dependency in `dtfs-central-api`. I assumed this was done for a reason, so made this query using the `createQueryBuilder` instead. If we are happy to install `typeorm` as a dependency in `dtfs-central-api`, then I can write this query in a more javascript-y way

Reply:
I obviously can't speak for the set up that has already happened, but my assumption would be it hasn't been added because it simply hasn't been needed yet and we should 100% utilise typeorm to make our db interactions as simple and tidy as possible 😄

Reply:
I have updated the way this query works now to not use the query builder. I'm not convinced it's easier to read, but let me know what you think!